send-data/aws-s3.mdx

```diff
@@ -62,9 +63,37 @@ def lambda_handler(event, context):
     if file_extension == '.csv':
         csv_data = csv.DictReader(io.StringIO(log_data))
         json_logs = list(csv_data)
-    elif file_extension == '.txt' or file_extension == '.log':
+    elif file_extension == '.txt':
         log_lines = log_data.strip().split("\n")
         json_logs = [{'message': line} for line in log_lines]
+    elif file_extension == '.log':
+        # IMPORTANT: Log files can be in various formats (JSON, XML, syslog, etc.)
+        try:
+            # First, try to parse as JSON (either one JSON object per line or a JSON array)
+            if log_data.strip().startswith('[') and log_data.strip().endswith(']'):
+                # Appears to be a JSON array
+                json_logs = json.loads(log_data)
+            else:
+                # Try parsing as NDJSON (one JSON object per line)
+                try:
+                    json_logs = ndjson.loads(log_data)
+                except:
+                    # If not valid NDJSON, check if each line might be JSON
+                    log_lines = log_data.strip().split("\n")
+                    json_logs = []
+                    for line in log_lines:
+                        try:
+                            # Try to parse each line as JSON
+                            parsed_line = json.loads(line)
+                            json_logs.append(parsed_line)
+                        except:
+                            # If not JSON, treat as plain text
+                            json_logs.append({'message': line})
```

Comment:
The line, even if plain text, needs to escape any internal quotes or other characters as it is being converted to a utf-8 quoted string so that it can be forwarded as a json log line. This would result in nginx or syslog data being malformed due to the implicit conversion not handling data that should be escaped...

Reply:
@darach fixed as well.